parser/SqlParser.ypp

```diff
@@ -759,6 +777,40 @@ opt_table_constraint_def_commalist:
     /* $$ = $1; */
   };
 
+opt_block_properties:
+  TOKEN_WITH TOKEN_BLOCKPROPERTIES '(' block_property_item_list ')' {
+    $$ = new quickstep::ParseBlockProperties(@1.first_line, @1.first_column, $4);
+  } | {
+    $$ = NULL;
+  };
+
+block_property_item_list:
+  block_property_item {
+    $$ = new quickstep::PtrList<quickstep::ParseBlockPropertyItem>();
+    $$->push_back($1);
+  }
+  | block_property_item_list ',' block_property_item {
+    $$ = $1;
+    $$->push_back($3);    
+  };
+
+block_property_item:
+  TOKEN_TYPE any_name {
+    $$ = new quickstep::ParseBlockPropertyItem(@1.first_line, @1.first_column, quickstep::ParseBlockPropertyItem::kType, $2);
+  }
+  | TOKEN_SORT any_name {
+    $$ = new quickstep::ParseBlockPropertyItem(@1.first_line, @1.first_column, quickstep::ParseBlockPropertyItem::kSort, $2);
+  }
+  | TOKEN_COMPRESS '(' name_commalist ')' {
+    $$ = new quickstep::ParseBlockPropertyItem(@1.first_line, @1.first_column, quickstep::ParseBlockPropertyItem::kCompress, $3);
+  } 
+  | TOKEN_COMPRESS TOKEN_ALL {
+    $$ = new quickstep::ParseBlockPropertyItem(@1.first_line, @1.first_column, 
+                    quickstep::ParseBlockPropertyItem::kCompress, 
```

Comment:
See this section of the style guide for rules on indenting parameters for function calls: https://google.github.io/styleguide/cppguide.html#Function_Calls